### PR TITLE
fixed issues with aks jit

### PR DESCRIPTION
**Pull request by kferrone**
## **Type**
enhancement

___

## **Description**
- Enhanced Kubernetes configuration handling in `jit.py` to conditionally include `certificate-authority-data` or `insecure-skip-tls-verify` based on the `K8Provider` value.
- Fixed an issue in `client.py` where the cache key generation could include slashes from the host URL, potentially leading to incorrect cache keys.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>jit.py</strong><dd><code>Enhance Kubernetes Configuration Handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/duplo_resource/jit.py

<li>Refactored <code>__k8s_exec_credential</code> and <code>__cluster_config</code> methods to <br>support conditional inclusion of <code>certificate-authority-data</code> or <br><code>insecure-skip-tls-verify</code>.<br> <li> Added a condition to include <code>certificate-authority-data</code> if <code>K8Provider</code> <br>is 0 and it exists.<br> <li> Added a condition to include <code>insecure-skip-tls-verify</code> if <code>K8Provider</code> is <br>1.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/52/files#diff-7e55cc23580ca0ed29a44b891d834a6cf3251a2a5a785f6291a4efe95b54cdf6">+15/-9</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>client.py</strong><dd><code>Fix Cache Key Generation in Client</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/duplocloud/client.py

<li>Modified <code>cache_key_for</code> method to ensure the host part of the URL does <br>not include slashes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/52/files#diff-ed921a2441e8df6f226f7024b875dea79afcd0d87e1326c37b19d232dba3f740">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/duplo_resource/jit.py [hunk 1]
```diff
@@ -120,15 +120,17 @@ def web(self):
     }
   
   def __k8s_exec_credential(self, creds):
+    c = {
```
Comment (akshay-duplo) [inline src/duplo_resource/jit.py:123]: can we give a meaningfull variable name here for `c`?

File: src/duplo_resource/jit.py [hunk 2]
```diff
@@ -139,12 +141,16 @@ def __k8s_exec_credential(self, creds):
   
   def __cluster_config(self, config):
     """Build a kubeconfig cluster object"""
+    c = {
```
Comment (akshay-duplo) [inline src/duplo_resource/jit.py:144]: same here

File: src/duplocloud/client.py [hunk 3]
```diff
@@ -456,7 +456,7 @@ def cache_key_for(self, name: str):
     Returns:
       The cache key as a string.
     """
-    h = self.host.split("://")[1]
+    h = self.host.split("://")[1].replace("/", "")
```
Comment (akshay-duplo) [inline src/duplocloud/client.py:459]: here as well.

**Review by akshay-duplo — CHANGES_REQUESTED** [review]

File: src/duplo_resource/jit.py (hunk 1 above)
Comment (kferrone) [reply src/duplo_resource/jit.py:123]: I suppose I could. I am a fan of single letter vars when the docs are good.

**Review by akshay-duplo — APPROVED** [review]
LGTM

**Pull request merged**